apps/ntag/ntagCmdApp/ntag_main.cpp

```diff
@@ -0,0 +1,428 @@
+	    blk[i] = reg;
+        }
+    }
+
+    if(le_hex_BinaryToString(blk, 6, blkHex, sizeof(blkHex)) != -1)
+        fprintf(stderr, "Session Regs of the tag is: %s\n", blkHex);
+    else
+        fprintf(stderr, "Failed in call to le_hex_BinaryToString of Session Regs \n");
+
+    dhubIO_PushString(TAG_MEM_SESSION_REG, DHUBIO_NOW, blkHex);
+}
+
+//--------------------------------------------------------------------------------------------------
+COMPONENT_INIT
+{
+    // calling ntag without arguments will print the Usage
+    if (le_arg_NumArgs() < 1)
+    {
+        Usage();
+        exit(EXIT_SUCCESS);
+    }
+    else
+    {
+        const char *commandPtr = le_arg_GetArg(0);
+
+        if ((0 == strcmp(commandPtr, "help")) ||
+            (0 == strcmp(commandPtr, "--help")) ||
+            (0 == strcmp(commandPtr, "-h")))
+        {
+            Usage();
+            exit(EXIT_SUCCESS);
+        }
+        else
+        {
+            if (strcmp(commandPtr, "readndef") == 0)
+            {
+                ReadTagExtended();
+            }
+            else if (strcmp(commandPtr, "clean") == 0)
+            {
+                CleanTag();
+            }
+            else if (strcmp(commandPtr, "writeall") == 0)
+            {
+                WriteTagMultipleRecords();
+            }
+            else if (strcmp(commandPtr, "write") == 0)
+            {
+                WriteTag();
+            }
+            else if (strcmp(commandPtr, "read") == 0)
+            {
+                ReadTag();
+            }
+            else if (strcmp(commandPtr, "erase") == 0)
+            {
+                EraseTag();
+            }
+            else if (strcmp(commandPtr, "pushhubmemory") == 0)
+            {
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_FORMATTED, DHUBIO_DATA_TYPE_BOOLEAN, ""));
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_SERIAL_NUM, DHUBIO_DATA_TYPE_STRING, ""));
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_MEM_CC, DHUBIO_DATA_TYPE_STRING, ""));
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_MEM_CFG_REG, DHUBIO_DATA_TYPE_STRING, ""));
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_MEM_SESSION_REG, DHUBIO_DATA_TYPE_STRING, ""));
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_MEM_EEPROM, DHUBIO_DATA_TYPE_STRING, ""));
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_MEM_SRAM, DHUBIO_DATA_TYPE_STRING, ""));
+                PushTagMemoryToDhub();
+            }
+	    else if (strcmp(commandPtr, "pushndef") == 0)
+            {
+                LE_ASSERT(LE_OK == dhubIO_CreateInput(TAG_NDEF, DHUBIO_DATA_TYPE_JSON, ""));
+		dhubIO_SetJsonExample(TAG_NDEF, JSON_EXAMPLE);
+                PushNdef();
+            }
+	    else if (strcmp(commandPtr, "text") == 0)
+            {
+                TextWrite(le_arg_GetArg(1));
+            }
+            else
+            {
+                Usage();
+                exit(EXIT_SUCCESS);
```

Comment:
This `exit` is unnecessary since the program will already exit successfully if allowed to continue.